modules/tts/tts-adapters/tts-adapter-cereproc/src/main/java/org/daisy/pipeline/tts/cereproc/impl/util/CereprocTTSUtil.java

```diff
@@ -0,0 +1,79 @@
+package org.daisy.pipeline.tts.cereproc.impl.util;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.Locale;
+import java.util.Optional;
+
+public class CereprocTTSUtil {
+
+    private Optional<Locale> locale;
+    RegexReplace regexReplace;
+    UCharReplacer charReplacer;
+
+    public CereprocTTSUtil(Optional<Locale> locale) {
+        this.locale = locale;
+        initRegexRules();
+        try {
+            initCharSubstitutionRules();
+        } catch (IOException e) {
+            throw new RuntimeException("Char substitution files not found");
+        }
+    }
+
+    private void initRegexRules() {
+        URL url;
+        String lang  = getCurrentLanguage();
+
+        if (lang.equals("sv")) {
+            url = CereprocTTSUtil.class.getResource("/regex/cereproc_sv.xml");
```

Comment:
This will not work in OSGi. Maybe you could use [`org.daisy.common.file.URLs.getResourceFromJAR`](http://daisy.github.io/pipeline/api/org/daisy/common/file/URLs.html#getResourceFromJAR-java.lang.String-java.lang.Class-)?

Reply:
OK! have have changed to getResource from jar